src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using EnvDTE;
+using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsProjectJsonToPackageReferenceMigrator))]
+    internal class VsProjectJsonToPackageReferenceMigrator : IVsProjectJsonToPackageReferenceMigrator
+    {
+        private readonly IVsSolutionManager _solutionManager;
+
+        [ImportingConstructor]
+        public VsProjectJsonToPackageReferenceMigrator(IVsSolutionManager solutionManager)
+        {
+            _solutionManager = solutionManager;
+        }
+        public IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectToPackageRef(string projectUniqueName)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            if(string.IsNullOrEmpty(projectUniqueName))
```

Comment:
We should move using `Assumes.Present` for all these null check. This API is provided via System.dll